comp/collector/collector/collectorimpl/collector.go

```diff
@@ -75,3 +104,255 @@ func newNoneCollector() provides {
+	}
+
+	c.checks[ch.ID()] = ch
+	c.notify(ch.ID(), collector.CheckRun)
+	return ch.ID(), nil
+}
+
+// StopCheck halts a check and remove the instance
+func (c *collectorImpl) StopCheck(id checkid.ID) error {
+	if !c.started() {
+		return fmt.Errorf("the collector is not running")
+	}
+
+	ch, found := c.get(id)
+	if !found {
+		return fmt.Errorf("cannot find a check with ID %s", id)
+	}
+
+	// unschedule the instance
+	err := c.scheduler.Cancel(id)
+	if err != nil {
+		return fmt.Errorf("an error occurred while canceling the check schedule: %s", err)
+	}
+
+	err = c.runner.StopCheck(id)
+	if err != nil {
+		// still attempt to cancel the check before returning the error
+		_ = c.cancelCheck(ch, c.cancelCheckTimeout)
+		return fmt.Errorf("an error occurred while stopping the check: %s", err)
+	}
+
+	err = c.cancelCheck(ch, c.cancelCheckTimeout)
+	if err != nil {
+		return fmt.Errorf("an error occurred while calling check.Cancel(): %s", err)
+	}
+
+	// remove the check from the stats map
+	expvars.RemoveCheckStats(id)
+
+	// vaporize the check
+	c.delete(id)
+
+	return nil
+}
+
+// cancelCheck calls Cancel on the passed check, with a timeout
+func (c *collectorImpl) cancelCheck(ch check.Check, timeout time.Duration) error {
+	done := make(chan struct{})
+
+	go func() {
+		ch.Cancel()
+		close(done)
+	}()
+
+	select {
+	case <-done:
+		return nil
+	case <-time.After(timeout):
+		return fmt.Errorf("timeout while calling check.Cancel() on check ID %s, timeout: %s", ch.ID(), timeout)
+	}
+}
+
+func (c *collectorImpl) get(id checkid.ID) (check.Check, bool) {
+	c.m.RLock()
+	defer c.m.RUnlock()
+
+	ch, found := c.checks[id]
+	return ch, found
+}
+
+// remove the check from the list
+func (c *collectorImpl) delete(id checkid.ID) {
+	c.m.Lock()
+	defer c.m.Unlock()
+
+	delete(c.checks, id)
+	c.notify(id, collector.CheckStop)
+}
+
+// lightweight shortcut to see if the collector has started
+func (c *collectorImpl) started() bool {
+	return c.state.Load() == started
+}
+
+// MapOverChecks call the callback with the list of checks locked.
+func (c *collectorImpl) MapOverChecks(cb func([]check.Info)) {
+	c.m.RLock()
+	defer c.m.RUnlock()
+
+	cInfo := []check.Info{}
+	for _, c := range c.checks {
+		cInfo = append(cInfo, c)
+	}
+	cb(cInfo)
+}
+
+// GetChecks copies checks
+func (c *collectorImpl) GetChecks() []check.Check {
+	c.m.RLock()
+	defer c.m.RUnlock()
+
+	chks := make([]check.Check, 0, len(c.checks))
+	for _, chck := range c.checks {
+		chks = append(chks, chck)
+	}
+
+	return chks
+}
+
+// GetAllInstanceIDs returns the ID's of all instances of a check
+func (c *collectorImpl) GetAllInstanceIDs(checkName string) []checkid.ID {
+	c.m.RLock()
+	defer c.m.RUnlock()
+
+	instances := []checkid.ID{}
+	for id, check := range c.checks {
+		if check.String() == checkName {
+			instances = append(instances, id)
+		}
+	}
+
+	return instances
+}
+
+// ReloadAllCheckInstances completely restarts a check with a new configuration
```

Comment:
Could you add a mention that the returned `checkid.ID` list is the killed list, and also do the same on the Component declaration. On first read I thought it might return the ids for the new checks.